docs/source/neighbors/all_neighbors.rst

```diff
@@ -0,0 +1,71 @@
+All-neighbors
+============
+
+All-neighbors is a specialized algorithm for building approximate all-neighbors k-NN graphs. Unlike traditional nearest neighbor indexes that are designed for searching, all-neighbors focuses on constructing complete k-NN graphs for entire datasets.
+
+This algorithm is particularly useful for:
+- Graph construction for visualization algorithms (UMAP, t-SNE)
+- Building connectivity graphs for clustering algorithms
+- Creating similarity graphs for graph-based machine learning
+- Batch processing of large datasets across multiple GPUs
+
+All-neighbors supports multiple underlying algorithms:
+- **Brute Force**: Exact nearest neighbors computation
+- **IVF-PQ**: Approximate nearest neighbors using inverted file with product quantization
+- **NN-Descent**: Approximate nearest neighbors using graph-based descent
+
+The algorithm partitions the dataset into clusters and distributes the work across multiple GPUs when possible, making it suitable for large-scale graph construction tasks.
+
+[ :doc:`C++ API <../cpp_api/neighbors_all_neighbors>` | :doc:`Python API <../python_api/neighbors_all_neighbors>` ]
```

Comment:
I think we should add the C API as well by adding a `neighbors_all_neighbors.rst` like we have for other index types here: https://github.com/rapidsai/cuvs/tree/branch-25.10/docs/source/c_api